lib/src/chord_parser.dart

```diff
@@ -13,9 +13,67 @@ class ChordProcessor {
     required String text,
     required TextStyle lyricsStyle,
     required chordStyle,
+    required widgetPadding,
     int transposeIncrement = 0,
   }) {
     final lines = text.split('\n');
+
+    bool _chordHasStartedOuter = false;
+    String _currentCharacters = '';
+    int _lastSpace = 0;
+    String _currentLine = '';
+    String _character = '';
+    double _media = MediaQuery.of(context).size.width;
+    int _characterIndex = 0;
+
+    //list to store our updated lines without overflows
+    final _newLines = <String>[];
+
+    //loop through the lines
+    for (var i = 0; i < lines.length; i++) {
+      _characterIndex = 0;
+      _currentCharacters = '';
+      _currentLine = lines[i];
+
+      //check if we have a long line
+      if (textWidth(_currentLine, lyricsStyle) >= _media) {
+        print('found a big line $_currentLine');
+        //work our way through the line and split when we need to
+        for (var j = 0; j < _currentLine.length; j++) {
+          _character = _currentLine[j];
+          if (_character == '[')
+            _chordHasStartedOuter = true;
+          else if (_character == ']')
+            _chordHasStartedOuter = false;
+          else if (_chordHasStartedOuter == true)
+            ;
+          else {
+            _currentCharacters += _character;
+            if (_character == ' ') {
+              //use this marker to only split where there are spaces. We can trim later.
+              _lastSpace = j;
+            }
+
+            //This is the point where we need to split
+            //widgetPadding has been added as a parameter to be passed from the build function
+            //It is intended to allow for padding in the widget when comparing it to screen width
+            //An additional buffer of around 10 might be needed to definitely stop overflow (ie. padding + 10).
+            if (textWidth(_currentCharacters, lyricsStyle) + widgetPadding >= _media) {
+              _newLines.add(_currentLine.substring(_characterIndex, _lastSpace).trim());
+              _currentCharacters = '';
+              _characterIndex = _lastSpace;
+            }
+          }
+        }
+        //add the rest of the long line
+        _newLines.add(_currentLine.substring(_characterIndex, _currentLine.length).trim());
+        
+      } else {
+        //otherwise just add the regular line
+        _newLines.add(_currentLine.trim());
+      }
+    }
+    
     List<ChordLyricsLine> _chordLyricsLines =
         lines.map<ChordLyricsLine>((line) {
```

Comment:
```suggestion
        _newLines.map<ChordLyricsLine>((line) {
```
I think this should be `_newLines`

Reply:
Haha yeah - copy and paste error there.